mujoco_warp/_src/collision_driver.py

```diff
@@ -60,8 +61,198 @@ def _zero_collision_arrays(
 
 
 @wp.func
-def _sphere_filter(
+def _plane_filter(
+  size1: float, size2: float, margin1: float, margin2: float, xpos1: wp.vec3, xpos2: wp.vec3, xmat1: wp.mat33, xmat2: wp.mat33
+) -> bool:
+  if size1 == 0.0:
+    # geom1 is a plane
+    dist = wp.dot(xpos2 - xpos1, wp.vec3(xmat1[0, 2], xmat1[1, 2], xmat1[2, 2]))
+    return dist <= size2 + wp.max(margin1, margin2)
+  elif size2 == 0.0:
+    # geom2 is a plane
+    dist = wp.dot(xpos1 - xpos2, wp.vec3(xmat2[0, 2], xmat2[1, 2], xmat2[2, 2]))
+    return dist <= size1 + wp.max(margin1, margin2)
+
+  return True
+
+
+@wp.func
+def _sphere_filter(size1: float, size2: float, margin1: float, margin2: float, xpos1: wp.vec3, xpos2: wp.vec3) -> bool:
+  bound = size1 + size2 + wp.max(margin1, margin2)
+  dif = xpos2 - xpos1
+  dist_sq = wp.dot(dif, dif)
+  return dist_sq <= bound * bound
+
+
+# TODO(team): improve performance by precomputing bounding box
+@wp.func
+def _aabb_filter(
+  # In:
+  center1: wp.vec3,
+  center2: wp.vec3,
+  size1: wp.vec3,
+  size2: wp.vec3,
+  margin1: float,
+  margin2: float,
+  xpos1: wp.vec3,
+  xpos2: wp.vec3,
+  xmat1: wp.mat33,
+  xmat2: wp.mat33,
+) -> bool:
+  """Axis aligned boxes collision.
+
+  references: see Ericson, Real-time Collision Detection section 4.2.
+              filterBox: filter contact based on global AABBs.
+  """
+  center1 = xmat1 @ center1 + xpos1
+  center2 = xmat2 @ center2 + xpos2
+
+  margin = wp.max(margin1, margin2)
+
+  max_x1 = -MJ_MAXVAL
+  max_y1 = -MJ_MAXVAL
+  max_z1 = -MJ_MAXVAL
+  min_x1 = MJ_MAXVAL
+  min_y1 = MJ_MAXVAL
+  min_z1 = MJ_MAXVAL
+
+  max_x2 = -MJ_MAXVAL
+  max_y2 = -MJ_MAXVAL
+  max_z2 = -MJ_MAXVAL
+  min_x2 = MJ_MAXVAL
+  min_y2 = MJ_MAXVAL
+  min_z2 = MJ_MAXVAL
+
+  sign = wp.vec2(-1.0, 1.0)
+
+  for i in range(2):
+    for j in range(2):
+      for k in range(2):
+        corner1 = wp.vec3(sign[i] * size1[0], sign[j] * size1[1], sign[k] * size1[2])
+        pos1 = center1 + xmat1 @ corner1
```

Comment:
nit: would it be an eensy weensy teeny tiny bit more performant if we remove the `center1 +` and `center2 + ` from the for loop and instead move them down to the overlap checks below?

Reply:
done